pkg/hydrate/controller.go

```diff
@@ -177,18 +171,34 @@ func (h *Hydrator) hydrate(sourceCommit, syncDir string) HydrationError {
 	return h.runHydrate(sourceCommit, syncDir)
 }
 
-// rehydrateOnError retries the hydration on errors.
-func (h *Hydrator) rehydrateOnError(sourceCommit, syncDir string) {
+// rehydrateOnError is triggered by the rehydrateTimer (every 30 mins)
+// It re-runs the rendering process when there is a previous error.
+func (h *Hydrator) rehydrateOnError() {
+	var sourceCommit, syncDir string
+	var hydrationErr HydrationError
 	errorFile := h.HydratedRoot.Join(cmpath.RelativeSlash(ErrorFile))
-	if _, err := os.Stat(errorFile.OSPath()); err != nil {
-		if !os.IsNotExist(err) {
-			klog.Warningf("unable to check the error file %s: %v", errorFile, err)
-		}
+	_, err := os.Stat(errorFile.OSPath())
+	switch {
+	case err != nil && os.IsNotExist(err):
+		// return directly if the error file doesn't exist.
 		return
+	case err != nil:
+		// Return a hydration error to retry next time
```

Comment:
Return? The error isn't being returned from this function... What's `complete` do with it?

Reply:
Updated the comment. `complete` creates a done file to indicate the hydration process is done, and also exports any errors to an error file.